utils/hct/hctdb.py

```diff
@@ -3599,62 +3391,52 @@ def UFI(name, **mappings):
             ],
             counters=("tex_store",),
         )
-        next_op_idx += 1
 
         # End of DXIL 1.2 opcodes.
-        self.set_op_count_for_version(1, 2, next_op_idx)
-        assert next_op_idx == 141, (
+        op_count = self.set_op_count_for_version(1, 2)
+        assert op_count == 141, (
```

Comment:
Would it be better to keep the op count argument to `set_op_count_for_version` and just have it assert internally that it matched `self.next_dxil_op_id`?

Reply:
That would be slightly easier, though it might also be easier for someone to update the count passed to the call, thinking that was the right way to adjust for a change that actually breaks DXIL, and a bit harder to notice in review. Having to update the number in the string as well makes it a bit more obvious that this isn't the right thing to do. I've caught exactly this change during review in at least one proposed addition in the past. Ultimately, I don't think one manually added assert per DXIL version is much of a burden for additional clarity.